crates/ln-dlc-node/src/node/invoice.rs

```diff
@@ -157,6 +167,18 @@ where
         route_hint_hop
     }
 
+    /// Inserts channel placeholder based on scid into the database
+    ///
+    /// This record will be updated once we intercept the HTLC and once the channel will be created.
+    fn insert_channel_from_intercept_id(
+        &self,
+        fake_scid: FakeScid,
+        counterparty: PublicKey,
+    ) -> Result<()> {
+        let channel = Channel::new(0, 0, counterparty, Some(fake_scid));
```

Comment:
🤔 That would create a channel in state `Pending`. Shouldn't we introduce a dedicated state for channels that are announced through a created invoice? I feel like we will get a lot of those records 🙈.

e.g. `Announced`

Reply:
That's reasonable I think.